src/upload-report/upload-report.test.tsx

```diff
@@ -261,17 +284,92 @@ describe('Upload Report', () => {
     expect(saveButton).not.toBeDisabled()
     userEvent.click(saveButton)
     await waitFor(() => {
-      expect(mockedOpenmrsFetch).toBeCalledTimes(3)
+      expect(mockedOpenmrsFetch).toBeCalledTimes(4)
     })
-    expect(mockedOpenmrsFetch.mock.calls[1][1].method).toBe('POST')
-    expect(mockedOpenmrsFetch.mock.calls[1][1].body).toBe(uploadFileRequestBody)
     expect(mockedOpenmrsFetch.mock.calls[2][1].method).toBe('POST')
-    expect(mockedOpenmrsFetch.mock.calls[2][1].body).toBe(
+    expect(mockedOpenmrsFetch.mock.calls[2][1].body).toBe(uploadFileRequestBody)
+    expect(mockedOpenmrsFetch.mock.calls[3][1].method).toBe('POST')
+    expect(mockedOpenmrsFetch.mock.calls[3][1].body).toBe(
       diagnosticReportRequestBody(new Date(currentDay).toISOString()),
     )
   })
 })
 
+it('should check for fhir diagnostic api call without having doctor uuid in request body on click of save and upload button for selecting self in doctors dropdown list', async () => {
```

Comment:
your test name should be crisp and represents user behaviour - it should not have test implementation details

`should save and upload report when user selects self in doctors dropdown and click save and upload button`

Reply:
yes